config/samples/test_shim_spin.yaml

```diff
@@ -17,6 +17,9 @@ spec:
     anonHttp:
       location: "https://github.com/spinframework/containerd-shim-spin/releases/download/v0.19.0/containerd-shim-spin-v2-linux-aarch64.tar.gz"
 
+  runtimeOptions:
+    SystemdCgroup: "true"
```

Comment:
So far, RCM only supports containerD-based clusters. This is because the nodeinstaller's implementation is limited (for now). However, technically, the nodeinstaller could handle, e.g., OpenShift, which is based on cri-o.

Up until now, I think the shim spec was very container runtime agnostic. However, with the introduction of `runtimeOption`, I'm not sure.

I'm not a CRI-O expert, but I can imagine that it requires different configurations or a different configuration format.

My question: Do we need to introduce a type to specify which runtime this configuration is for?

I'm aware that this is very hypothetical, especially since the CRD is in an alpha state and we have never talked in depth about supporting open shift until now. But I'm nevertheless curious about your thoughts.

Reply:
thats a good point. I could see a few options:
1. update this to `containerdRuntimeOptions` or something else containerd specific
2. in the future if we support a cri-o shim, we could add a field called `runtimeType` that has the option `containerd` or `crio`. This informs how to use `runtimeOptions`
3. We make a breaking change when the time comes